sw/device/lib/dif/dif_lc_ctrl.h

```diff
@@ -0,0 +1,403 @@
+   */
+  kDifLcCtrlAlertCorrupt,
+} dif_lc_ctrl_alert_t;
+
+/**
+ * Creates a new handle for the lifecycle controller.
+ *
+ * This function does not actuate the hardware.
+ *
+ * @param params Hardware instantiation parameters.
+ * @param[out] lc Out param for the initialized handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_lc_ctrl_result_t dif_lc_ctrl_init(dif_lc_ctrl_params_t params,
+                                      dif_lc_ctrl_t *lc);
+
+/**
+ * Returns the current state of the lifecycle controller.
+ *
+ * @param lc A lifecycle handle.
+ * @param[out] state Out-param for the controller's state.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_lc_ctrl_result_t dif_lc_ctrl_get_state(const dif_lc_ctrl_t *lc,
+                                           dif_lc_ctrl_state_t *state);
+
+/**
+ * Returns the number of lifecycle transitions that this device has attempted,
+ * up to 16.
+ *
+ * @param lc A lifecycle handle.
+ * @param[out] state Out-param for the controller's state.
+ * @return The result of the operation.
+ */
+// Open Q: do we want a separate error type for when the counter is all-ones? I
+// think it should just be part of kError.
+DIF_WARN_UNUSED_RESULT
+dif_lc_ctrl_result_t dif_lc_ctrl_get_attempts(const dif_lc_ctrl_t *lc,
+                                              uint8_t *count);
+
+/**
+ * Returns the current status of the lifecycle controller.
+ *
+ * @param lc A lifecycle handle.
+ * @param[out] status Out-param for the controller's status.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_lc_ctrl_result_t dif_lc_ctrl_get_status(const dif_lc_ctrl_t *lc,
+                                            dif_lc_ctrl_status_t *status);
+
+/**
+ * Returns the current personalization state of the lifecycle controller.
+ *
+ * @param lc A lifecycle handle.
+ * @param[out] state Out-param for the controller's personalization state.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_lc_ctrl_result_t dif_lc_ctrl_get_id_state(const dif_lc_ctrl_t *lc,
+                                              dif_lc_ctrl_id_state_t *state);
+
+/**
+ * Forces a particular alert, causing it to be escalated as if the hardware had
+ * raised it.
+ *
+ * @param lc A lifecycle handle.
+ * @param alert The alert to force.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_lc_ctrl_result_t dif_lc_ctrl_alert_force(const dif_lc_ctrl_t *lc,
+                                             dif_lc_ctrl_alert_t alert);
+
+/**
+ * Attempts to acquire the lifecycle controller's HW mutex.
+ *
+ * Returns `kDifLcCtrlMutexHeld` if acquisition fails. It is recommended to
+ * call this function in a busy loop to acquire the mutex.
+ *
+ * @param lc A lifecycle handle.
+ * @return The result of the operation.
+ */
+// Open Q: do we want to be checking REGWEN for all operations dependent on the
+// mutex?
```

Comment:
All operations accessing a register that is protected by regwen should check regwen before accessing that register, and if it's 0, return a locked error (as with other DIFs).

@msfschaffner On the HW side, the `TRANSITION_REGWEN` register docs seem to say that this is usually 0x1, and sometimes 0x0 if there's a transition in progress, or if TAP has claimed the mutex. This goes against the docs on `claim_transition_if`, which says that the value is only 1 when the mutex is claimed, and 0 otherwise. Clarity on this interface would be appreciated. From the software side, we'd prefer that `TRANSITION_REGWEN` was 0 unless the mutex was claimed.

Reply:
Good point, thanks for raising that. I'll clarify that in the docs and update the HW such that it behaves the way you suggest (I would also prefer it to behave that way).

Reply:
See #4364